shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -362,18 +363,21 @@ private void doInitialFlutterViewRun() {
       // So this is expected behavior in many cases.
       return;
     }
-
+    String initialRoute = host.getInitialRoute();
+    if (initialRoute == null) {
+      initialRoute = maybeGetInitialRouteFromIntent(host.getActivity().getIntent());
+    }
     Log.v(
         TAG,
         "Executing Dart entrypoint: "
             + host.getDartEntrypointFunctionName()
             + ", and sending initial route: "
-            + host.getInitialRoute());
+            + initialRoute);
 
     // The engine needs to receive the Flutter app's initial route before executing any
     // Dart code to ensure that the initial route arrives in time to be applied.
-    if (host.getInitialRoute() != null) {
-      flutterEngine.getNavigationChannel().setInitialRoute(host.getInitialRoute());
+    if (initialRoute != null) {
+      flutterEngine.getNavigationChannel().setInitialRoute(initialRoute);
```

Comment:
I think this has a new semantic now (of the default route being null whereas getInitialRoute used to be non-null). I think it's actually consequential downstream even though it looks like a platform channel https://github.com/flutter/engine/blob/5772deea53a2d21aca05c3c5c91bfd5f2068f88b/shell/common/engine.cc#L322.

Reply:
I did a null check before i hand it to the message channel, it will be not null after that point, no?

Reply:
No not after the call. I meant before. host.getInitialRoute() previously never returns null as far as I can tell. Now there are opportunities to never call flutterEngine.getNavigationChannel().setInitialRoute

Reply:
ah you are right, I didn't notice the previous if check does not actually do anything.

Reply:
right. And it's consequential since initial route is "special" in that it plugs to a custom piece of memory in the engine and the dart:ui window API connects to that memory directly.